### PR TITLE
fix: support chain switching for walletconnect

### PR DESCRIPTION
**Pull request by woodenfurniture**
Updates WalletConnect v2 implementation to support chain switching.
See https://github.com/shapeshift/web/pull/5425 for details.

### CODE REVIEW

File: packages/hdwallet-walletconnectV2/src/walletconnectV2.ts [hunk 1]
```diff
@@ -358,4 +358,22 @@ export class WalletConnectV2HDWallet implements HDWallet, ETHWallet {
...
+  }
+
+  public async ethSwitchChain({ chainId }: AddEthereumChainParameter): Promise<void> {
+    const parsedChainId = parseInt(chainId, 16);
+    if (isNaN(parsedChainId) || this.chainId === parsedChainId) {
+      return;
+    }
+
+    this.chainId = parsedChainId;
+    this.provider.chainId = parsedChainId;
```
Comment (gomesalexandre) [inline packages/hdwallet-walletconnectV2/src/walletconnectV2.ts:373]: Do we need to mutate this optimistically? `this.provider` being the Ethereum provider, its `chainId` property should automagically reflect the one after the switch?

File: packages/hdwallet-walletconnectV2/src/walletconnectV2.ts [hunk 2]
```diff
@@ -358,4 +358,22 @@ export class WalletConnectV2HDWallet implements HDWallet, ETHWallet {
...
+    return this.provider.chainId;
+  }
+
+  public async ethSwitchChain({ chainId }: AddEthereumChainParameter): Promise<void> {
+    const parsedChainId = parseInt(chainId, 16);
+    if (isNaN(parsedChainId) || this.chainId === parsedChainId) {
+      return;
+    }
+
+    this.chainId = parsedChainId;
```
Comment (gomesalexandre) [inline packages/hdwallet-walletconnectV2/src/walletconnectV2.ts:372]: We may want to not be optimistic here and move this after the promise execution

File: packages/hdwallet-walletconnectV2/src/walletconnectV2.ts (hunk 1 above)
Comment (woodenfurniture) [reply packages/hdwallet-walletconnectV2/src/walletconnectV2.ts:373]: excellent point, actioning now

**Review by gomesalexandre — DISMISSED** [review]
Tested with https://github.com/shapeshift/web/pull/5425 - stamp pending lerna version bumpy

**Review by gomesalexandre — APPROVED** [review]

**Pull request merged**